ibis/expr/types/json.py

```diff
@@ -98,6 +100,132 @@ def __getitem__(
         """
         return ops.JSONGetItem(self, key).to_expr()
 
+    def unwrap_as(self, dtype: dt.DataType | str) -> ir.Value:
+        """Unwrap JSON into a specific data type.
+
+        Returns
+        -------
+        Value
+            An Ibis expression of a more specific type than JSON
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> data = {
+        ...     "jstring": ['"a"', '""', None, "null"],
+        ...     "jbool": ["true", "false", "null", None],
+        ...     "jint": ["1", "null", None, "2"],
+        ...     "jfloat": ["42.42", None, "null", "37.37"],
+        ...     "jmap": ['{"a": 1}', "null", None, "{}"],
+        ...     "jarray": ["[]", "null", None, '[{},"1",2]'],
+        ... }
+        >>> t = ibis.memtable(data, schema=dict.fromkeys(data.keys(), "json"))
+        >>> t
+        ┏━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┳━━━┓
+        ┃ jstring              ┃ jbool                ┃ jint                 ┃ … ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━╇━━━┩
+        │ json                 │ json                 │ json                 │ … │
+        ├──────────────────────┼──────────────────────┼──────────────────────┼───┤
+        │ 'a'                  │ True                 │ 1                    │ … │
+        │ ''                   │ False                │ None                 │ … │
+        │ NULL                 │ None                 │ NULL                 │ … │
+        │ None                 │ NULL                 │ 2                    │ … │
+        └──────────────────────┴──────────────────────┴──────────────────────┴───┘
+        >>> t.select(unwrapped=t.jstring.unwrap_as(str), original=t.jstring)
```

Comment:
Should ``.unwrap_as(str)`` in this actually be ``.unwrap_as("string")``, or are both valid as it's a dtype?

Reply:
Both are valid. The `dt.dtype` call will normalize inputs.